### PR TITLE
Display event

### PR DESCRIPTION
**Pull request by dipscoder**
Display:
- User's Upcoming Events
- New Events
And
- Made route for individual Event dynamic

@Abhishek-kumar09

### CODE REVIEW

**Comment by Abhishek-kumar09** [comment]
The card is not having perfect margins:
image

If there are no bookmarked events then remove that. If search functionality is not woking yet, comment the code for that.

**Comment by Abhishek-kumar09** [comment]
Link checkout buttons for the events to checkout to that event.

**Comment by dipscoder** [comment]
> Link checkout buttons for the events to checkout to that event.

Its already completed

**Comment by dipscoder** [comment]
@Abhishek-kumar09  Please have a look

**Comment by Abhishek-kumar09** [comment]
image

Proper padding is necessary.
The cards are having improper padding.
It should not have fixed width or if so try to get a consistent design.

**Comment by Abhishek-kumar09** [comment]
image
The design is breaking

**Comment by dipscoder** [comment]
> image
> The design is breaking

Everything is working fine in my local

Screenshot (355)

File: src/components/UserNewEvents.js [hunk 1]
```diff
@@ -131,10 +131,22 @@ function UserNewEvents() {
...
+                ) : (
+                  ''
+                )
+              }
               <Button
-                style={ event?.eventName.length <= 11 ? {marginTop: '21px', backgroundColor: 'white'} : {marginTop: '1px',backgroundColor: 'white'} }
+                style={event?.eventName.length <= 11 ? { marginTop: '21px', backgroundColor: 'white' } : { marginTop: '1px', backgroundColor: 'white' }}
                 className={classes.checkOut}
-                onClick={() => history.push(`/events/${eventID[idx]}`)}
+                // onClick={() => history.push(`/events/${eventID[idx]}`)}
```
Comment (Abhishek-kumar09) [inline src/components/UserNewEvents.js]: isko revert kar dena

**Review by Abhishek-kumar09 — COMMENTED** [review]
The code was not working  for the non logged in users. 
Have you tried it?
It was showing null error on user which was faced by many other contributors.

I did some changes in package file and I added it in this PR. Please give a look correct errors if any.

**Comment by dipscoder** [comment]
> 
> 
> The code was not working for the non logged in users.
> Have you tried it?
> It was showing null error on user which was faced by many other contributors.
> 
> I did some changes in package file and I added it in this PR. Please give a look correct errors if any.

Just because right now, Profile can only be viewed by logged in users
Current route for profile: "/profile"

I will make this dynamic so that anyone can view any other user's profile 
Updated route will be: " /profile/firstname-lastname-4digitsOfUserID"

Is it Ok if I use this route? 
Also, should I create a new PR for this or add changes in current PR only?

File: src/components/UserNewEvents.js (hunk 1 above)
Comment (dipscoder) [reply src/components/UserNewEvents.js]: Ok

**Comment by dipscoder** [comment]
@Abhishek-kumar09  please have a look.

**Comment by Abhishek-kumar09** [comment]
Everything is working fine right, with or without logIn?

**Comment by Abhishek-kumar09** [comment]
Actually, I have not checked it locally, so confirming your side.

**Comment by dipscoder** [comment]
Yes, everything is working fine

**Pull request merged**